packages/react/src/Pillar/Pillar.tsx

```diff
@@ -0,0 +1,130 @@
+
+/**
+ * Main stylesheet (as a CSS Module)
+ */
+import styles from './Pillar.module.css'
+import {Icon as IconProps} from '@primer/octicons-react'
+
+export const PillarIconColors = Colors
+
+export const defaultPillarIconColor = PillarIconColors[0]
+export type PillarProps<C extends keyof JSX.IntrinsicElements = 'div'> = React.HTMLAttributes<C> & {
+  /**
+   * The HTML element used to render the grid.
+   */
+  as?: C | 'div' | 'span' | 'article'
+  /**
+   * Valid children include Pillar.Image, Pillar.Heading, and Pillar.Description
+   */
+  children:
+    | React.ReactNode
+    | React.ReactElement<PillarIconProps>
+    | React.ReactElement<PillarHeadingProps>
+    | React.ReactElement<PillarDescriptionProps>
+  /**
+   * Aligns the pillar content
+   */
+  align?: 'start' | 'center'
+} & (C extends 'span'
+    ? BaseProps<HTMLSpanElement>
+    : C extends 'article'
+    ? BaseProps<HTMLElement>
+    : BaseProps<HTMLDivElement>)
+
+const PillarRoot = forwardRef(
+  (
+    {children, className, as = 'div', align = 'start', ...rest}: PropsWithChildren<PillarProps>,
+    ref: Ref<HTMLDivElement>
+  ) => {
+    const filteredChildren = React.Children.toArray(children).filter(child => {
+      if (React.isValidElement(child) && typeof child.type !== 'string') {
+        if (child.type === PillarIcon || child.type === PillarHeading || child.type === PillarDescription) {
+          return true
+        }
+      }
+      return false
+    })
+
+    const validElements = ['div', 'span', 'article']
+    const Component = validElements.includes(as) ? as : 'div'
+
+    return (
+      <Component
+        className={clsx(styles.Pillar, className, styles[`Pillar--align-${align}`])}
+        ref={ref}
+        {...(rest as HTMLAttributes<HTMLElement>)}
+      >
+        {filteredChildren}
+      </Component>
+    )
+  }
+)
+
+type PillarIconProps = BaseProps<HTMLSpanElement> & {
+  icon: React.ReactNode | IconProps
+  color?: typeof PillarIconColors[number]
+}
+
+function PillarIcon({icon: Icon, className, color = defaultPillarIconColor, ...rest}: PillarIconProps) {
+  return (
+    <span className={clsx(styles.Pillar__icon, styles[`Pillar__icon--color-${color}`], className)} {...rest}>
+      {typeof Icon === 'function' ? <Icon size={24} /> : React.isValidElement(Icon) && React.cloneElement(Icon)}
+    </span>
+  )
+}
+
+type PillarHeadingProps = BaseProps<HTMLHeadingElement> & {
+  children: React.ReactNode | React.ReactNode[]
+  as?: Exclude<HeadingTags['as'], 'h1'>
+} & HeadingProps
+
+const PillarHeading = forwardRef<HTMLHeadingElement, PillarHeadingProps>(
+  ({children, as = HeadingTags[2], size = '6', className, ...rest}, ref) => {
+    return (
+      <Heading size={size} className={clsx(styles.Pillar__heading, className)} ref={ref} as={as} {...rest}>
+        {children}
+      </Heading>
+    )
+  }
+)
+
+type PillarDescriptionProps = BaseProps<HTMLParagraphElement> & {
+  children: React.ReactNode | React.ReactNode[]
+}
```

Comment:
Could we extend TextProps here too? In case someone needs to customise certain values like variant or `as`

Reply:
I'd like to hold on providing such props.

- I can't think of a case where we need to modify the description to an html that isn't a paragraph via `as` props. 
- For variants and other font styles this is similar to the `Card` component description, where we agreed with site designers to keep it simple to one color/weight/size variant, until a strong use case or a design need pops up that confronts that.